pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -191,3 +193,41 @@ func getBlockVolumeToHostMap(ctx context.Context, cMgr *common.Manager,
 	}
 	return volumeIDNodeUUIDMap, nil
 }
+
+// getVolumeManagerForVolumeID returns volume manager for the given volumeId
+// if multi-vcenter-csi-topology feature is disabled legacy volume manager is removed
+// from controller.manager.VolumeManager
+// if multi-vcenter-csi-topology feature is enabled but volumeInfoService is not instantiated volume manager
+// is returned from controller.managers.VolumeManagers for controller.managers.CnsConfig.Global.VCenterIP
+// if multi-vcenter-csi-topology feature is enabled and volumeInfoService is instantiated volume manager is returned for
+// the vCenter IP mapped to the requested VolumeID. vCenter for the requested volume is obtained from volumeInfoService
+func getVolumeManagerForVolumeID(ctx context.Context, controller *controller, volumeId string,
+	volumeInfoService cnsvolumeinfo.VolumeInfoService) (cnsvolume.Manager, error) {
+	log := logger.GetLogger(ctx)
+	var volumeManager cnsvolume.Manager
+	var volumeManagerfound bool
+	if multivCenterCSITopologyEnabled {
+		if volumeInfoService != nil {
```

Comment:
Can we use `len(c.managers.VcenterConfigs) > 1` instead?

Reply:
done. it is more clear with this check.